### PR TITLE
[BottomNavigation]! Remove encoding/decoding behavior for custom properties

### PR DESCRIPTION
**Pull request by jverkoey**
Closes https://github.com/material-components/material-components-ios/issues/3939

### CODE REVIEW

**Comment by jverkoey** [comment]
@jmdetloff were the tests added in 5c88af59d152dfb45782486a6b784048f720f85a related specifically to coding? If not, I'll move them to a separate file so that they're not deleted in this PR.

File: components/BottomNavigation/src/private/MDCBottomNavigationItemView.m [hunk 1]
```diff
@@ -98,53 +82,11 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
              @"Unexpected number of subviews. Expected %lu but restored %lu. Unarchiving may fail.",
```
Comment (romoore) [inline components/BottomNavigation/src/private/MDCBottomNavigationItemView.m:82]: Why are we keeping the `-initWithCoder:` code above? I thought we're dropping support for NSCoding?

**Review by romoore — DISMISSED** [review]

**Comment by romoore** [comment]
@jverkoey Those tests should be kept, they are not using NSCoding and are just exercising the new layout code.

**Review by ianegordon — DISMISSED** [review]
Could you confirm with a comment that there are property setters handling any state changes for all appropriate keys that are being removed?  (To handle the KVO properly as we are no longer going to be explicitly decoding)

File: components/BottomNavigation/src/private/MDCBottomNavigationItemView.m (hunk 1 above)
Comment (jverkoey) [reply components/BottomNavigation/src/private/MDCBottomNavigationItemView.m:82]: As best I can tell the above code doesn't have to do with our custom property coding/decoding behavior. Am I looking at it wrong?

**Comment by jverkoey** [comment]
> Could you confirm with a comment that there are property setters handling any state changes for all appropriate keys that are being removed? (To handle the KVO properly as we are no longer going to be explicitly decoding)

I'm not certain I understand what you're referring to here. These PRs don't affect property implementations. Can you give me an example of the specific case you feel this PR is affecting?

**Comment by jverkoey** [comment]
@romoore Restored the tests.

**Comment by jverkoey** [comment]
This PR is still ready for review.

**Comment by jverkoey** [comment]
Gentle ping.

File: components/BottomNavigation/tests/unit/BottomNavigationItemViewTests.m [hunk 2]
```diff
@@ -35,64 +35,11 @@ @interface MDCBottomNavigationItemView (Testing)
 @property(nonatomic, strong) UILabel *label;
 @end
 
-@interface BottomNavigationItemViewCodingTests : XCTestCase
+@interface BottomNavigationItemViewTests : XCTestCase
 
 @end
```
Comment (codeman7) [inline components/BottomNavigation/tests/unit/BottomNavigationItemViewTests.m:40]: Is this an extra `@end`?

**Review by codeman7 — APPROVED** [review]

File: components/BottomNavigation/tests/unit/BottomNavigationItemViewTests.m (hunk 2 above)
Comment (jverkoey) [reply components/BottomNavigation/tests/unit/BottomNavigationItemViewTests.m:40]: Nope - it's closing the BottomNavigationItemViewTests interface declaration.

**Pull request merged**